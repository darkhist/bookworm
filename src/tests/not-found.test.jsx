import React from 'react';
import { shallow, mount } from 'enzyme';

import NotFound from '../components/notFound/notFound';

describe('404', () => {
  it('should render without crashing', () => {
    const notFound = shallow(<NotFound />);
    expect(notFound).toMatchSnapshot();
  });

  it('should render a 404 error message', () => {
    const notFound = shallow(<NotFound />);
    expect(notFound.html()).toContain('404');
    expect(notFound.html()).toContain('Page not found');
  });

  it('should render a btn that redirects home', () => {
    const notFound = mount(<NotFound />);
    const btn = notFound.find('a');
    expect(btn.exists()).toBeTruthy();
    expect(btn.text()).toBe('Go Back');
  });
});
