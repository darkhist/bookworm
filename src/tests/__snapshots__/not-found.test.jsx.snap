// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 should render without crashing 1`] = `
<styled.div>
  <styled.h1>
     404 
  </styled.h1>
  <styled.p>
     Page not found 
  </styled.p>
  <styled.button
    as="a"
    href="/"
    type="button"
  >
    Go Back
  </styled.button>
</styled.div>
`;
